### PR TITLE
Deprecate copy task

### PR DESCRIPTION
**Pull request by karajas**
@chcosta @maririos

### CODE REVIEW

**Comment by maririos** [comment]
I swear I put comments here :( but either way, trying it again.

File: buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json [hunk 1]
```diff
@@ -235,30 +259,58 @@
...
       "timeoutInMinutes": 0,
+      "refName": "CopyFiles1",
       "task": {
-        "id": "1d341bb0-2106-458c-8422-d00bcea6512a",
+        "id": "5bfb729a-a7c8-4a78-a7c3-8d717bb7c13c",
+        "versionSpec": "2.*",
+        "definitionType": "task"
+      },
+      "inputs": {
+        "SourceFolder": "$(PB_GitDirectory)",
```
Comment (maririos) [inline buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json]: Is it save to overwrite the source folder that the task was using before? Will we get the same content?

File: buildpipeline/DotNet-CoreClr-Trusted-Linux.json [hunk 2]
```diff
@@ -199,30 +221,58 @@
...
+      "refName": "PublishBuildArtifacts2",
+      "task": {
+        "id": "2ff763a7-ce83-4e1f-bc89-0ae63477cebe",
         "versionSpec": "1.*",
         "definitionType": "task"
       },
       "inputs": {
-        "CopyRoot": "$(DockerCopyDest)",
+        "PathtoPublish": "$(Build.StagingDirectory)\\BuildLogs",
         "Contents": "**/*.log",
```
Comment (maririos) [inline buildpipeline/DotNet-CoreClr-Trusted-Linux.json]: I haven't seen a field called `Contents` in the Publish task. Did u test it?

File: buildpipeline/DotNet-CoreClr-Trusted-Mac.json [hunk 3]
```diff
@@ -91,22 +101,48 @@
...
+      "refName": "CopyFiles1",
       "task": {
-        "id": "1d341bb0-2106-458c-8422-d00bcea6512a",
+        "id": "5bfb729a-a7c8-4a78-a7c3-8d717bb7c13c",
+        "versionSpec": "2.*",
+        "definitionType": "task"
+      },
+      "inputs": {
+        "SourceFolder": "$(PB_GitDirectory)",
+        "Contents": "**/*.log",
```
Comment (maririos) [inline buildpipeline/DotNet-CoreClr-Trusted-Mac.json]: You are overwriting the `Contents` property. I guess the question for the source applies here too

File: buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json (hunk 1 above)
Comment (karajas) [reply buildpipeline/DotNet-CoreClr-Trusted-Linux-Crossbuild.json]: sorry copy paste error, fixing now.

**Pull request merged**